Missed review comments for EventCriteria, making a `hasCriteria` meth…

CodeDrivenMitch opened the pull request:
…od on the EventCriteria interface

smcvb reviewed (commented):
Looks nice, but I am missing a test case for the new method :-)

CodeDrivenMitch commented:
@smcvb You are right. Added tests. Also added a test for the flatten method, which exposed a bug in the `FilteredEventCriteria`. Resolved that as well.

smcvb reviewed (approved):
My concerns have been addressed, hence I'm approving this pull request.
